include/xtensor-r/rcontainer.hpp

```diff
@@ -222,10 +222,10 @@ namespace xt
      */
     template <class D, template <class> class SP>
     template <class S>
-    inline void rcontainer<D, SP>::resize(S&& shape)
+    inline void rcontainer<D, SP>::resize(S&& shape, bool force)
     {
         // if SEXP not initialized, it will be NULL (e.g. in constructor)
-        if (Rf_isNull(*this) || shape.size() != this->dimension() || !std::equal(std::begin(shape), std::end(shape), this->shape().cbegin()))
+        if (Rf_isNull(*this) || shape.size() != this->dimension() || !std::equal(std::begin(shape), std::end(shape), this->shape().cbegin()) || force)
```

Comment:
more as a general question/comment, can short-circuiting work if `force` is the last option? I think not.